tutorials/dynamic-programming.md

```diff
@@ -10,12 +10,124 @@ keywords:
   - algorithm
 ---
 
-<!-- TODO: change the format. See arrays.md -->
+<TutorialAuthors names="@Cyber-Machine"/>
 
-### Suggested Problems
+## Overview
 
-* [0072 - Edit Distance (Hard)](../solutions/0000-0099/edit-distance-hard)
-* [0877 - Stone Game (Medium)](../solutions/0800-0899/stone-game-medium)
-* [1510 - Stone Game IV (Hard)](../solutions/1500-1599/stone-game-iv-hard)
-* [1692 - Count Ways to Distribute Candies (Hard)](../solutions/1600-1699/count-ways-to-distribute-candies-hard)
-* [2110 - Number of Smooth Descent Periods of a Stock (Medium)](../solutions/2100-2199/number-of-smooth-descent-periods-of-a-stock-medium)
+If we want to generate all possible solutions to a particular problem we can use recursion or brute force, but this is often not efficient for solving problems with larger input size.
+
+If the problem can be broken down into **overlapping and independent subproblems**, then we can use Dynamic Programming to minimize the cost associated with repetitive computations when compared to brute force algorithms by **solving and storing sub-problems** and using it to solve the original problem.
+
+Let's take a look at Leetcode problem [70. Climbing Stairs](https://leetcode.com/problems/climbing-stairs/).
+
+> You are climbing a staircase. It takes **n steps** to reach the top.
+>
+> Each time you can either **climb 1 or 2 steps**. In how many **distinct** ways can you climb to the top?
+>
+
+#### Solution-1 : Recursion 
+We can count number of ways to reach end stairs using recursion, in order to reach nth step we have two options either take (n-1)th stair or (n-2)th stair.
+
+
+i.e. stair(n) = stair(n-1) + stair(n-2) 
```

Comment:
I think it would be better to apply LaTex on `stair(n) = stair(n - 1) + stair(n - 2)`.